### PR TITLE
ade7913 driver improvements

### PR DESCRIPTION
**Pull request by nalajcie**
- fix support for 3x ADE invocation
- fix EDMA per-channel functions

## Description
NIL v33

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: adc/ade7913/ade7913.c [hunk 1]
```diff
@@ -247,10 +247,9 @@ int ade7913_sync(oid_t *device, const char *cs, int devcnt, int snap)
 	oid_t gpiodev;
 	platformctl_t pctl;
 
-	int muxes[4] = { pctl_mux_gpio_ad_18, pctl_mux_gpio_ad_19,
-		pctl_mux_gpio_ad_20, pctl_mux_gpio_ad_29 };
+	int muxes[4] = { pctl_mux_gpio_ad_29, pctl_mux_gpio_ad_18, pctl_mux_gpio_ad_19, pctl_mux_gpio_ad_20 };
 
-	int pins[4] = { 17, 18, 19, 28 };
+	int pins[4] = { 28, 17, 18, 19 };
```
Comment (agkaminski) [inline adc/ade7913/ade7913.c:252]: Perhaps take the opportunity to move this to board_config.h?

File: adc/ade7913/ade7913-driver.c [hunk 2]
```diff
@@ -52,6 +52,8 @@
 #define log_error(fmt, ...) do { printf(LOG_TAG COL_RED fmt COL_NORMAL "\n", ##__VA_ARGS__); } while (0)
 /* clang-format on */
 
+/* enable debugging of DMA IRQ not in-sync processing */
+#define DEBUG_NOTSYNC 1
```
Comment (agkaminski) [inline adc/ade7913/ade7913-driver.c:56]: I'm not sure about committing a code with debug enabled
```suggestion
/* enable debugging of DMA IRQ not in-sync processing */
#define DEBUG_NOTSYNC 0
```

File: adc/ade7913/ade7913.c (hunk 1 above)
Comment (nalajcie) [reply adc/ade7913/ade7913.c:252]: there is too much hard-coded logic in irq handler `edma_spi_rcv_irq_handler` -  (raw access to GPIO port with the assumption all pins are on a single one, etc.) - I don't want to generalize it too much (adding multiple GPIO ports/for's, etc.) and risk the ISR will take longer than expected.

File: adc/ade7913/ade7913-driver.c (hunk 2 above)
Comment (nalajcie) [reply adc/ade7913/ade7913-driver.c:56]: I need this for NIL project and don't want every user to ask to enable it manually.

**Review by agkaminski — APPROVED** [review]

**Pull request merged**